src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -467,8 +467,15 @@ private int CheckAndAdjustPositionForSpeculativeAttribute(int position)
 
         protected override IOperation GetOperationCore(SyntaxNode node, CancellationToken cancellationToken)
         {
+            if (!Root.FullSpan.Contains(node.FullSpan))
+            {
+                // given node must be sub node of this root.
+                return null;
```

Comment:
since I can access Root I might move this check outside of SemanticModel and put it in Operation and leave the design to you guys.

current behavior is, with that check removed, implementation of GetOperation will throw at some point since given input node is wrong, and assert and GetOperation will return null since it catches all exception and return null.

http://source.roslyn.io/#Microsoft.CodeAnalysis/Compilation/SemanticModel.cs,82